src/main/java/seedu/address/ui/CommandHistory.java

```diff
@@ -0,0 +1,110 @@
+package seedu.address.ui;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.logging.Logger;
+
+import seedu.address.commons.core.LogsCenter;
+
+/**
+ * A class to encapsulate the history of commands entered by the user.
+ */
+public class CommandHistory {
+    private static final Logger logger = LogsCenter.getLogger(CommandHistory.class);
+    private final List<String> history;
+    private int pointer;
+
+    /**
+     * Constructor for the history of commands entered.
+     *
+     * @param history
+     */
+    private CommandHistory(List<String> history) {
+        logger.info(history.toString());
+        this.history = history;
+        pointer = history.size();
+    }
+
+    public static CommandHistory init() {
+        return new CommandHistory(new ArrayList<String>());
+    }
+
+    /**
+     * Adds the given command to the history of commands.
+     *
+     * @param command
+     */
+    public void add(String command) {
+        requireAllNonNull(command);
+        history.add(command);
+        pointer = history.size();
+    }
+
+    /**
+     * Gets the next command in history.
+     *
+     * @return The next command.
+     */
+    public String getNext() {
+        if (hasNext()) {
+            return history.get(moveForward());
+        } else {
+            throw new NoSuchElementException();
+        }
```

Comment:
Can we get the command box to clear the very first time there's no next element? (So like it shouldn't clear if I'm just typing and accidentally press down, but if i'm going through history and press down after the most recently used command it should clear) I think that's more consistent with CLI behavior.